pkg/validation/usage_groups_test.go

```diff
@@ -94,36 +88,54 @@ func TestUsageGroupConfig_GetUsageGroups(t *testing.T) {
 		{
 			Name:     "no_labels",
 			TenantID: "tenant1",
-			Config: UsageGroupConfig{
-				config: map[string][]*labels.Matcher{
-					"app/foo": testMustParseMatcher(t, `{service_name="foo"}`),
-				},
+			Config: map[string]string{
+				"app/foo": `{service_name="foo"}`,
 			},
 			Labels: phlaremodel.Labels{},
 			Want: UsageGroupMatch{
 				tenantID: "tenant1",
+				names:    []string{},
 			},
 		},
 		{
 			Name:     "disjoint_labels_do_not_match",
 			TenantID: "tenant1",
-			Config: UsageGroupConfig{
-				config: map[string][]*labels.Matcher{
-					"app/foo": testMustParseMatcher(t, `{namespace="foo", container="bar"}`),
-				},
+			Config: map[string]string{
+				"app/foo": `{namespace="foo", container="bar"}`,
 			},
 			Labels: phlaremodel.Labels{
 				{Name: "service_name", Value: "foo"},
 			},
 			Want: UsageGroupMatch{
 				tenantID: "tenant1",
+				names:    []string{},
+			},
+		},
+		{
+			Name:     "dynamic_usage_group_names",
+			TenantID: "tenant1",
+			Config: map[string]string{
+				"app/${labels.service_name}": `{service_name=~"(.*)"}`,
+			},
+			Labels: phlaremodel.Labels{
+				{Name: "service_name", Value: "foo"},
+			},
+			Want: UsageGroupMatch{
+				tenantID: "tenant1",
+				names: []string{
+					"app/foo",
+				},
 			},
```

Comment:
I think we also should handle empty label value like missing label name (unlikely empty labels make makes it that far). Here roughly the test case:

```suggestion
			},
		},	
		{
			Name:     "dynamic_usage_group_names_missing_label",
			TenantID: "tenant1",
			Config: map[string]string{
				"app/${labels.service_name}/${labels.env}": `{service_name=~"(.*)"}`,
			},
			Labels: phlaremodel.Labels{
				{Name: "service_name", Value: "foo"},
			},
			Want: UsageGroupMatch{
				tenantID: "tenant1",
				names: []string{},
			},
		},	
		{
			Name:     "dynamic_usage_group_names_empty_label",
			TenantID: "tenant1",
			Config: map[string]string{
				"app/${labels.service_name}": `{service_name=~"(.*)"}`,
			},
			Labels: phlaremodel.Labels{
				{Name: "service_name", Value: ""},
			},
			Want: UsageGroupMatch{
				tenantID: "tenant1",
				names: []string{},
			},
		},				
```